File: insertionSort/iSTest.py
import unittest
from insertionSort import insertion_sort

class TestStringMethods( unittest.TestCase ):

    def test_numbers( self ):
        self.assertEqual( insertion_sort([ 3, 4, 1, 2 ]), [ 1, 2, 3, 4 ] )

    def test_words( self ):
        self.assertEqual( insertion_sort([ 'cat', 'bat', 'hat' ]), [ 'bat',
            'cat', 'hat' ] )

    def test_characters( self ):
        self.assertEqual( insertion_sort([ 'c', 'f', 'z', 'd' ]), [ 'c', 'd',
            'f', 'z' ] )


if __name__ == '__main__':
    unittest.main()

File: insertionSort/iSTest.js
var expect = require('chai').expect;
var insertionSort = require('./insertionSort');

describe('Insertion sort', function() {
  it('Should sort an array of numbers', function() {
    expect(insertionSort([ 3, 4, 2, 1 ])).to.eql([ 1, 2, 3, 4 ])
  });

  it('Should sort an array of words', function() {
    expect(insertionSort([ 'cat', 'bat', 'hat' ])).to.eql([ 'bat', 'cat', 'hat' ]);
  });

  it('Should sort an array of characters', function() {
    expect(insertionSort([ 'c', 'x', 'f', 'd' ])).to.eql([ 'c', 'd', 'f', 'x' ]);
  });
});
